src/transformers/generation/logits_process.py

```diff
@@ -336,7 +336,7 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         # if score < 0 then repetition penalty has to be multiplied to reduce the token probabilities
         score = torch.where(score < 0, score * self.penalty, score / self.penalty)
 
-        scores.scatter_(1, input_ids, score)
+        scores = scores.scatter(1, input_ids, score)
         return scores
```

Comment:
nit: we should do `return scores_processed` or `return scores` in ALL processors, for the sake of keeping a consistent pattern